src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -42,58 +55,104 @@
  * execution times significantly if applicable.
  * <p>
  */
-public class CommonTermsQueryBuilder extends QueryBuilder implements BoostableQueryBuilder<CommonTermsQueryBuilder> {
+public class CommonTermsQueryBuilder extends QueryBuilder<CommonTermsQueryBuilder> implements BoostableQueryBuilder<CommonTermsQueryBuilder> {
 
     public static final String NAME = "common";
 
+    public static final float DEFAULT_CUTOFF_FREQ = 0.01f;
+
+    public static final Operator DEFAULT_HIGH_FREQ_OCCUR = Operator.OR;
+
+    public static final Operator DEFAULT_LOW_FREQ_OCCUR = Operator.OR;
+
+    public static final boolean DEFAULT_DISABLE_COORD = true;
+
     public static enum Operator {
-        OR, AND
+        OR, AND;
+
+        public Occur toMustOrShouldClause() {
```

Comment:
I think I'd leave this out of the enum, also would prefer a switch so that we have the OR case, the AND case, and we throw error otherwise (rather than doing AND and assume everything else is OR)